app/src/main/java/com/mapbox/maps/testapp/examples/OfflineActivity.kt

```diff
@@ -0,0 +1,329 @@
+        }
+      }
+      expected.error?.let { tileRegionError ->
+        logErrorMessage("TileRegionError: $tileRegionError")
+      }
+    }
+    offlineManager.getAllStylePacks { expected ->
+      if (expected.isValue) {
+        expected.value?.let { stylePackList ->
+          logInfoMessage("Existing style packs: $stylePackList")
+        }
+      }
+      expected.error?.let { stylePackError ->
+        logErrorMessage("StylePackError: $stylePackError")
+      }
+    }
+  }
+
+  private fun removeOfflineRegions() {
+    // Remove the tile region with the tile region ID.
+    // Note this will not remove the downloaded tile packs, instead, it will just mark the tileset
+    // not a part of a tile region. The tiles still exists as a predictive cache in TileStore.
+    TileStore.getInstance().removeTileRegion(TILE_REGION_ID)
+
+    // Set the disk quota to zero, so that tile regions are fully evicted
+    // when removed. The TileStore is also used when `ResourceOptions.isLoadTilePacksFromNetwork`
+    // is `true`, and also by the Navigation SDK.
+    // This removes the tiles from the predictive cache.
+    TileStore.getInstance().setOption(TileStoreOptions.DISK_QUOTA, Value(0))
+
+    // Remove the style pack with the style url.
+    // Note this will not remove the downloaded style pack, instead, it will just mark the resources
+    // not a part of the existing style pack. The resources still exists as ambient cache.
+    if (this::offlineManager.isInitialized) {
+      offlineManager.removeStylePack(Style.OUTDOORS)
+    }
+    // Remove the existing style resources from ambient cache using cache manager.
+    val cacheManager = CacheManager(
+      MapInitOptions.getDefaultResourceOptions(this)
+    )
+    cacheManager.clearAmbientCache {}
+
+    // Reset progressbar.
+    updateStylePackDownloadProgress(0)
+    updateTileRegionDownloadProgress(0)
+  }
+
+  private fun updateStylePackDownloadProgress(progress: Int, message: String? = null) {
+    style_pack_download_progress.progress = progress
+    message?.let {
+      offlineLogsAdapter.addLog(OfflineLog.StylePackProgress(it))
+    }
+  }
+
+  private fun updateTileRegionDownloadProgress(progress: Int, message: String? = null) {
+    tile_pack_download_progress.progress = progress
+    message?.let {
+      offlineLogsAdapter.addLog(OfflineLog.TilePackProgress(it))
+    }
+  }
+
+  private fun logInfoMessage(message: String) {
+    offlineLogsAdapter.addLog(OfflineLog.Info(message))
+  }
+
+  private fun logErrorMessage(message: String) {
+    offlineLogsAdapter.addLog(OfflineLog.Error(message))
+  }
+
+  private fun logSuccessMessage(message: String) {
+    offlineLogsAdapter.addLog(OfflineLog.Success(message))
+  }
+
+  override fun onStart() {
+    super.onStart()
+    mapView?.onStart()
+  }
+
+  override fun onStop() {
+    super.onStop()
+    mapView?.onStop()
+  }
+
+  override fun onDestroy() {
+    super.onDestroy()
```

Comment:
Guess we should cancel the downloads that in progress.

Reply:
The `removeOfflineRegions()` below should delete the regions downloaded and cancel any ongoing downloads.